packages/runtime/src/plugins/useUpstreamRetry.ts

```diff
@@ -0,0 +1,174 @@
+          retryDelay = 1000,
+          shouldRetry = ({ response, executionResult }) => {
+            if (response) {
+              // If network error or rate limited, retry
+              if (
+                response.status >= 500 ||
+                response.status === 429 ||
+                response.headers.get('Retry-After')
+              ) {
+                return true;
+              }
+            }
+            // If there are errors that are not original GraphQL errors, retry
+            if (
+              !isAsyncIterable(executionResult) &&
+              executionResult.errors?.length &&
+              !executionResult.errors.some(isOriginalGraphQLError)
+            ) {
+              return true;
+            }
+            return false;
+          },
+        } = optsForReq;
+        if (maxRetries > 0) {
+          setExecutor(function (executionRequest: ExecutionRequest) {
+            let retries = maxRetries + 1;
+            let executionResult: MaybeAsyncIterable<ExecutionResult>;
+            function retry(): MaybePromise<
+              MaybeAsyncIterable<ExecutionResult>
+            > {
+              retries--;
+              try {
+                if (retries < 0) {
+                  return executionResult;
+                }
+                const requestTime = Date.now();
+                return mapMaybePromise(
+                  executor(executionRequest),
+                  (currRes) => {
+                    executionResult = currRes;
+                    let retryAfterSeconds: number | undefined;
+                    const response =
+                      executionRequestResponseMap.get(executionRequest);
+                    const retryAfterHeader =
+                      response?.headers.get('Retry-After');
+                    if (retryAfterHeader) {
+                      retryAfterSeconds = parseInt(retryAfterHeader);
+                      if (isNaN(retryAfterSeconds)) {
+                        const dateTime = new Date(retryAfterHeader).getTime();
+                        if (!isNaN(dateTime)) {
+                          retryAfterSeconds = dateTime - requestTime;
+                        }
+                      }
+                    }
+                    let currentRetryDelay: number | undefined;
+                    if (retryAfterSeconds) {
+                      currentRetryDelay = retryAfterSeconds * 1000;
+                    }
+                    if (
+                      shouldRetry({
+                        executionRequest,
+                        executionResult,
+                        response,
+                      })
+                    ) {
+                      currentRetryDelay ||= retryDelay;
+                      return new Promise((resolve) => {
+                        const timeout = setTimeout(() => {
+                          timeouts.delete(timeout);
+                          resolve(retry());
+                        }, retryDelay);
```

Comment:
The `retryDelay` is used instead of the computed `currentRetryDelay` variable.

```suggestion
                        }, currentRetryDelay);
```

Reply:
Done!